### PR TITLE
Replace @enumOfCaseObject with @simpleEnum

### LINKED ISSUE #284: integrate with new ZIO Schema `simpleEnum` annotation
version 0.4.14 of ZIO Schema introduces the `simpleEnum` annotation - see PR https://github.com/zio/zio-schema/pull/564

We need to replace the existing custom `enumOfCaseObjects` annotation with this new standard one and update the codecs, examples and docs.

Also see https://github.com/zio/zio-dynamodb/issues/137 for the original issue for replacing custom annotations with standard ones

### PR DESCRIPTION
**Pull request by grouzen**
/fixes #284

### CODE REVIEW

File: dynamodb/src/test/scala/zio/dynamodb/codec/ItemDecoderSpec.scala [hunk 1]
```diff
@@ -284,35 +284,20 @@ object ItemDecoderSpec extends ZIOSpecDefault with CodecTestFixtures {
 
       assert(actual)(isRight(equalTo(PreBilled(id = 1, s = "foobar"))))
     },
-    test("decodes case object only enum with @enumOfCaseObjects annotation and without @caseName annotation") {
+    test("decodes case object only enum with @simpleEnum annotation and without @caseName annotation") {
```
Comment (grouzen) [inline dynamodb/src/test/scala/zio/dynamodb/codec/ItemDecoderSpec.scala:287]: I'm not pretty sure if we need to keep these test cases as it seems they test what was tested by zio-schema.

File: dynamodb/src/test/scala/zio/dynamodb/codec/ItemDecoderSpec.scala [hunk 2]
```diff
@@ -284,35 +284,20 @@ object ItemDecoderSpec extends ZIOSpecDefault with CodecTestFixtures {
...
-      assert(actual)(
-        isLeft(
-          hasMessage(
-            equalTo(
-              "Can not decode enum String(ONE) - @enumOfCaseObjects annotation present when all instances are not case objects."
-            )
-          )
-        )
-      )
-    },
```
Comment (grouzen) [inline dynamodb/src/test/scala/zio/dynamodb/codec/ItemDecoderSpec.scala:315]: Using  `@simpleEnum` checks this in compile time.

File: dynamodb/src/test/scala/zio/dynamodb/codec/ItemEncoderSpec.scala [hunk 3]
```diff
@@ -219,35 +218,21 @@ object ItemEncoderSpec extends ZIOSpecDefault with CodecTestFixtures {
...
-        isFailure(
-          hasMessage(
-            equalTo(
-              "Can not encode enum ONE - @enumOfCaseObjects annotation present when all instances are not case objects."
-            )
-          )
-        )
-      )
-    },
-    test("encodes enum and honours @caseName annotation when there is no @enumOfCaseObjects annotation") {
```
Comment (grouzen) [inline dynamodb/src/test/scala/zio/dynamodb/codec/ItemEncoderSpec.scala:250]: Using  `@simpleEnum` checks this in compile time.

**Review by googley42 — APPROVED** [review]
LGTM - many thanks for your contribution!

**Pull request merged**